### DISCUSSION TITLE
Option to delay audio for a Media Player part of a Group Player

### DISCUSSION THREAD

**Discussion by Verguldebarman** [music-assistant/support#3318] [Q&A] [answered]
I like the Group Player option very much. It has unfortunately one big issue: 

--> **unable to sync/delay individual Media Players**

Therefore the Group Player option is useless in case the Media Players are out of sync. That produces a lot of unwanted echo in the audio....

Similar to the [Group Delay Correction](https://support.google.com/chromecast/answer/6318642) option in Google Chromecast devices.
![Screenshot_2024-12-18-09-42-15-64_2d2bd67b5e15ae98c151ac739cd6881e.jpg](https://github.com/user-attachments/assets/78f7fb20-c89e-4d81-ba7e-1ef9dc25f3b4)

**Comment by OzGav** [comment]
If you use the native MA players they play in sync. There will be some HA players in the future (integration dependent) that will also play in sync

**Comment by Verguldebarman** [comment]
Yeah, I guess MA plays in sync, however...the speakers (Media Players) play the music out of sync ;-)

MA cannot currently account for distance (or individual speaker delays, caching, etc.). Therefore, you need a delay option per the Media Player part of the Player Group.

In my case I have created a Player Group with the following (native) MA Media Players:
- 1x Chomecast Utra
- 2x Google mini's

There is an unfortunate delay between the Chromecast Ultra and the mini's..... the Utra is 60ms ahead.

**Reply by OzGav** [reply]
Cast works differently. That link you posted above would be the solution I would expect

**Reply by OzGav** [reply]
This may also be of interest https://music-assistant.io/faq/tech-info/#player-perfect-sync

**Reply by Verguldebarman** [reply]
Yeah, found that and theoretically it should work fine indeed! 

However, I can confirm that there is a huge delay in the Speaker Group as configured. 

Also delaying the Google speakers as per the link does not work, as this is native Google Home functionality. And as the document describes; I must use native MA speakers in order to "sync perfectly".

**Reply by OzGav** [reply]
I just provided that as background info. As per the MA docs sync correction has to be done via the Google Home App.

**Comment by marcelveldt** [comment, marked as answer]
Cast players need to be grouped in Google home not in MA!
